client/src/page/Menu.tsx

```diff
@@ -1,35 +1,33 @@
 import * as React from 'react';
 import { useTheme } from '@mui/material/styles';
 import Box from '@mui/material/Box';
-import { useDispatch, useSelector } from 'react-redux';
-import { closeMenu, openMenu } from 'store/menu.slice';
-import { RootState } from 'store/store';
+import useAppState from 'store/AppAccess';
 import MenuToolbar from './MenuToolbar';
 import DrawerComponent from './DrawerComponent';
 
-const drawerWidth = 240;
+const drawerwidth = 240;
 
 const hooks = () => {
   const theme = useTheme();
-  const menuState = useSelector((state: RootState) => state.menu);
-  const dispatch = useDispatch();
+  const { state: menuState, actions: menuAction } = useAppState();
   const [anchorElUser, setAnchorElUser] =
     React.useState<HTMLButtonElement | null>(null);
-  return { theme, menuState, dispatch, anchorElUser, setAnchorElUser };
+  return { theme, menuState, menuAction, anchorElUser, setAnchorElUser };
 };
 
 function MiniDrawer() {
-  const { theme, menuState, dispatch, anchorElUser, setAnchorElUser } = hooks();
+  const { theme, menuState, menuAction, anchorElUser, setAnchorElUser } =
+    hooks();
   const handleCloseUserMenu = () => setAnchorElUser(null);
   const handleOpenUserMenu = (event: React.MouseEvent<HTMLButtonElement>) =>
     setAnchorElUser(event.currentTarget);
-  const handleDrawerOpen = () => dispatch(openMenu());
-  const handleDrawerClose = () => dispatch(closeMenu());
+  const handleDrawerOpen = () => menuAction.open();
```

Comment:
It looks like you are essentially wrapping the dispatch function in menuAction.open(). If you are, the dispatch is a standard method in redux terminology. Has this consistency been maintained at some other place?

Reply:
We've made sure to only use dispatch in the `Access` modules, thus keeping Redux "at an arm's length". I'm not sure if this answers your question or if you want us to take further action here?

Reply:
All the redux code is in `store/` I guess. Does any other part of the codebase knows about redux? If not, your statement about keeping Redux "at an arm's length" is valid. Then the changes made here make sense.

I am a bit puzzled about completely wrapping the redux. You can probably keep the reducer, action, dispatch hidden. But how do you keep the subscribe hidden? The react redux components need to be used across the codebase, at least in the AssetBoard / Cart part of the codebase. Is it not?

Reply:
No other part of the codebase knows about Redux, so we should be good.

Since we pass the state through the `Access` module as a Redux state, we can simply subscribe to it directly like this:
```typescript
import useCart from 'store/CartAccess';

function ShoppingCart() {
  const { state, actions } = useCart();
}
```

It works perfectly fine for our use case, but there may be other features of Redux needed in the future, that might prove difficult to utilize. But for now, we've encountered no issues.

Reply:
The subscription (state?) is handled by the same hook that provides the wrapped actions. 
We made it this way to have a single point of contact between the store and the rest of the application. 
The state of the slice (part of the complete state) is provided by the useAppSelector in the *Access.ts files.
It this instance, the state of the menu is provided by 

const state = useAppSelector((store) => store.menu);

in client/src/store/AppAccess.ts